### PR TITLE
Migration initializer part I: discover migratable resources

### PR DESCRIPTION
**Pull request by caesarxuchao**
This one can be review separately from other pulls.

The functions are for the `migrator-intializer` in KEP https://github.com/kubernetes/community/blob/master/keps/sig-api-machinery/0030-storage-migration.md#alpha-workflow.

### CODE REVIEW

File: pkg/initializer/discover.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	"github.com/golang/glog"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/discovery"
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
+)
+
+type disovery struct {
```
Comment (yliaog) [inline pkg/initializer/discover.go]: typo

File: pkg/initializer/discover.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	disocveryClient  discovery.ServerResourcesInterface
```
Comment (yliaog) [inline pkg/initializer/discover.go]: typo, please fix "disocveryClient"

File: pkg/initializer/discover.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		return nil, err
+	}
+	resourceToGroupVersions := make(map[string][]schema.GroupVersion)
+	resourceLists, err := d.disocveryClient.ServerResources()
+	if err != nil {
+		return nil, err
+	}
+	for _, resourceList := range resourceLists {
+		gv, err := schema.ParseGroupVersion(resourceList.GroupVersion)
+		if err != nil {
```
Comment (yliaog) [inline pkg/initializer/discover.go]: print out the error?

**Comment by caesarxuchao** [comment]
Fixed. PTAL. Thanks.

**Pull request merged**